### DISCUSSION TITLE
data-pagefind-body cannot contain attribute value?

### DISCUSSION THREAD

**Discussion by dprothero** [Pagefind/pagefind#152] [Q&A]
I'm trying to add the `data-pagefind-body` attribute to an element in my code, but since it's React, it renders it as `data-pagefind-body="true"`. It seems Pagefind doesn't work with this, is that correct?

**Comment by bglw** [comment]
Hmm no, Pagefind isn't affected by that any value of that attribute. Relevant code here:
https://github.com/CloudCannon/pagefind/blob/bd1d93141fa41d382171feaf1ac8f06082fa9a36/pagefind/src/fossick/parser.rs#L148

Are the pages that Pagefind is indexing fully SSR rendered? My guess would be that the attribute is only being added on React's client hydration

**Reply by justsilencia** [reply]
@bglw I'm having an issue with pagefind not limiting its index to elements with the data-pagefind-body attribute as well. I'm using astro for SSG. I haven't even enabled SSR in astro.config, so everything is SSG as of now. However, I'm using mdx for markdown. And these are where the data-pagefind-body attributes are being placed (in an article tag). I'm not using any React components in the mdx yet, so hydration shouldn't be an issue. Pagefind is simply indexing the entire page and ignoring the attributes. Could the use of mdx be causing this? I suppose my next step will be to try with standard md files and see if the issue persists.

**Reply by bglw** [reply]
@justsilencia Hmm that shouldn't be an issue. Do you have a sample repo or code snippet that you could share that I could look into? Even just an output HTML file that isn't being indexed correctly would be great 🙂

**Reply by justsilencia** [reply]
@bglw I will be working on this project again tomorrow and I'll try to reproduce it in a manner that you can examine. Otherwise if I figure it out I'll let you know what the issue was. Thanks a ton for the prompt response!! This project is looking great.

**Reply by bglw** [reply]
@dprothero & @justsilencia:

I've just done a quick and dirty test using [slorber/docusaurus-starter](https://github.com/slorber/docusaurus-starter), where I dropped in a `data-pagefind-body` onto [line 17 of src/pages/index.js](https://github.com/slorber/docusaurus-starter/blob/7ff544f779c3773a01b3aa55b6009aa990efdbc6/src/pages/index.js#L17) such that only that element on the homepage should be indexed. 

I do then see React output this as `data-pagefind-body="true"` in the static HTML, and everything works as expected. Running Pagefind I see:
```
[Parsing files]
Found a data-pagefind-body element on the site.
↳ Ignoring pages without this tag.

[Reading languages]
Discovered 1 language: en

[Building search indexes]
Total:
  Indexed 1 language
  Indexed 1 page
  Indexed 4 words
  Indexed 0 filters
  Indexed 0 sorts
```

Searching `Tutorial` gives me the homepage, and no other pages return for any search query. Removing the `data-pagefind-body` attribute correctly indexes all pages.

Not sure what your issue(s) could then be, but it must be something else hiding in these sites. If either of you do get any small reproduction of the issue you can share I'd be eager to take a look — sorry that I can't dig in any further, but let me know 🙂🙏

**Reply by justsilencia** [reply]
@bglw Okay because I was able to reproduce this issue in a sample repository, I went ahead and posted a new issue with all of the relevant information.  I added a link to the repo and to the site which is published on Netlify, as well as the pagefind output found in the Netlify build. Thanks

**Reply by bglw** [reply]
@dprothero I have released a fix for @justsilencia in #163. In case your situation is also resulting from these floating elements outside the `html`, this might now be working for you in [Pagefind v0.10.4](https://github.com/CloudCannon/pagefind/releases/tag/v0.10.4). Let me know! If not, send through some more info if/when you get a chance and I'll take another look 🙂
